src/typecheck/operation.rs

```diff
@@ -378,6 +379,25 @@ pub fn get_nop_type(
             vec![mk_uniftype::str(), mk_uniftype::num(), mk_uniftype::num()],
             mk_uniftype::str(),
         ),
+        // Dyn -> Dyn -> Dyn -> Dyn -> Dyn
+        NAryOp::SealRecordTail() => (
+            vec![
+                mk_uniftype::dynamic(),
+                mk_uniftype::dynamic(),
+                mk_uniftype::dynamic(),
+                mk_uniftype::dynamic(),
+            ],
+            mk_uniftype::dynamic(),
+        ),
+        // Dyn -> Dyn -> Dyn -> Dyn
+        NAryOp::UnsealRecordTail() => (
+            vec![
+                mk_uniftype::dynamic(),
+                mk_uniftype::dynamic(),
+                mk_uniftype::dynamic(),
+            ],
+            mk_uniftype::dynamic(),
+        ),
```

Comment:
I imagine we could have slightly more precise types here, but it's probably quite useless.

Reply:
Yeah, the conclusion I came to is that types for these operations aren't worth enforcing as any error would have to be (1) an error within the stdlib, which should be caught by the test suite or (2) a result of users misusing primitive ops, in which case runtime errors are probably sufficient.

That said, out of curiosity, what types do you think would be appropriate here? I tried a few ideas but wasn't happy with any of them.

Taking `%record_unseal_tail%` as an example, my thinking process was: 

- the sealing key and label are typed as `Dyn` everywhere, so we don't have much choice there,
- we can't use row types since we'd have to somehow represent the realization of the row variable into its constituent fields & types,
- we could use dictionary types, but since these operations are designed to work on row types (1) that felt a little misleading and (2) the only valid value type would be `Dyn` which doesn't provide much value.

I may well be missing something though - happy to hear any ideas!

Reply:
I was thinking of `{_ : Dyn}`, mostly.

> we could use dictionary types, but since these operations are designed to work on row types (1) that felt a little misleading and (2) the only valid rhs type would be Dyn which doesn't seem to provide much value.

Mh I wouldn't say those operations "are designed to work on row types". Those primops act on data, and they are pretty dynamic operations moving studd around in the representation of the record. The question is what shape of data they expect and return: we can say at least that most of the arguments and the return type are records, so `{_ : Dyn}`.

I agree it's not very useful.

Reply:
I think you've actually convinced me that `{ _ : Dyn }` makes slightly more sense than just `Dyn`, so I'll update it before I merge. :sweat_smile: